### PR TITLE
fetch please check

### PR DESCRIPTION
**Pull request by Yasin-Goenendi**
fix the fetch problem and adress the api

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -15,4 +15,14 @@ const nextConfig = {
...
 };
 
-module.exports = nextConfig;
+module.exports = {
+  nextConfig,
+  images: {
+    remotePatterns: [
+      {
+        protocol: 'https',
+        hostname: '**.org',
```
Comment (Kheirah) [inline next.config.js:24]: It's not a prudent strategy to forgo on security by allowing every subdomain of `.org`. Allow only that which is actually accessed by your app.

**Comment by Marcel-FR** [comment]
I can't preview your app on Vercel. The data mapping should solve this though.

File: pages/index.js [hunk 2]
```diff
@@ -1,13 +1,65 @@
 import styled from "styled-components";
+import { useEffect, useState } from "react";
+import Image from "next/image";
+
+
+
+
```
Comment (Kheirah) [inline pages/index.js:7]: What are those empty code lines for? Please use prettier for formatting your code. It helps with readability. Use it in all your files before requesting a review.

File: pages/index.js [hunk 3]
```diff
@@ -1,13 +1,65 @@
...
+.then(response => response.json())
+.then(response => {
+  console.log(response);
+  setData(response);
+})
+.catch(err => {
+  console.error(err);
+})
+
+  })
```
Comment (Kheirah) [inline pages/index.js:35]: `useEffect` is missing its second argument.

**Review by Kheirah — CHANGES_REQUESTED** [review]
Good start!

But your Vercel deployment isn't working, yet. Your local project isn't working either. You should not submit a PR for a review when your project is broken. I left some comments to help get your project fixed. Take a look and adjust your code where necessary.

Also, the description of your PR shouldn't be empty. We don't know what you're implementing without the info of what you're basing your implementation on. Be descriptive and leave a link to the user story of that PR.

File: pages/index.js [hunk 4]
```diff
@@ -2,64 +2,45 @@ import styled from "styled-components";
...
-.catch(err => {
-  console.error(err);
-})
-
-  })
+    async function fetchMovies() {
+      const apiKey = process.env.API_KEY;
+      const response = await fetch(
+        `https://api.themoviedb.org/3/movie/popular?api_key=9b1dc3e910cce9037e3a21dbbd912753&language=en-US&page=1`
+      );
```
Comment (Kheirah) [inline pages/index.js:13]: You're reading the key from the environment object, but you're not using that information in your fetch. Instead, you're passing the secret key itself. Better use the `apiKey` here.

File: pages/index.js [hunk 5]
```diff
@@ -2,64 +2,45 @@ import styled from "styled-components";
...
+      const data = await response.json();
+      setMovies(data.results);
+      console.log(data);
```
Comment (Kheirah) [inline pages/index.js:16]: Remove `console.logs`.

**Review by Kheirah — CHANGES_REQUESTED** [review]
Nice work!

Have a look at the code line concerning the api key and adjust your code accordingly to keep the secret actually secret.

File: pages/index.js [hunk 6]
```diff
@@ -7,13 +7,12 @@ function MoviesList() {
 
   useEffect(() => {
     async function fetchMovies() {
-      const apiKey = process.env.API_KEY;
+      const apiKey = `9b1dc3e910cce9037e3a21dbbd912753`;
```
Comment (Kheirah) [inline pages/index.js:10]: Not quite!
Reading the secret from your environment object was correct.

File: pages/index.js [hunk 7]
```diff
@@ -7,13 +7,12 @@ function MoviesList() {
...
       const response = await fetch(
         `https://api.themoviedb.org/3/movie/popular?api_key=9b1dc3e910cce9037e3a21dbbd912753&language=en-US&page=1`
```
Comment (Kheirah) [inline pages/index.js:12]: This is not correct. You need to pass the secret to your string via interpolation.

**Review by Kheirah — CHANGES_REQUESTED** [review]
You're not there, yet =)

File: pages/index.js [hunk 8]
```diff
@@ -22,17 +23,18 @@ function MoviesList() {
...
-              src={`https://image.tmdb.org/t/p/w500${movie.poster_path}`}
-              width="300"
-              height="300"
-              alt={movie.title}
-            />
-          </div>
-        ))}
+        {movies?.length > 0 &&
+          movies.map((movie) => (
+            <div key={movie.id} className="movie">
```
Comment (Kheirah) [inline pages/index.js:28]: What is the class `movie` doing? You should use `styled components` to style your html tags.

**Review by Kheirah — CHANGES_REQUESTED** [review]
Check your dependencies if in fact you do need them. Further make sure to create the same environment information.

File: package.json [hunk 9]
```diff
@@ -10,8 +10,11 @@
     "test": "jest --watchAll"
   },
   "dependencies": {
+    "dotenv": "^16.0.3",
+    "dotenv-webpack": "^8.0.1",
     "eslint": "8.35.0",
     "eslint-config-next": "13.2.3",
+    "fs": "^0.0.1-security",
```
[the remark this answers is not in the document]
Comment (Yasin-Goenendi) [reply package.json:17]: i thought there had been another problem, but there wasn't

File: pages/index.js (hunk 8 above)
Comment (Yasin-Goenendi) [reply pages/index.js]: i fix it

File: package.json [hunk 10]
```diff
@@ -16,6 +16,7 @@
     "eslint-config-next": "13.2.3",
     "fs": "^0.0.1-security",
```
Comment (Kheirah) [inline package.json:17]: I still doubt whether you need that package `fs`.

**Review by Kheirah — APPROVED** [review]

**Pull request merged**